templates/optimizers.yaml

```diff
@@ -39,55 +40,12 @@ spec:
               data: |
-                import os
-                from zquantum.optimizers.utils import save_optimization_results
-                from zquantum.core.cost_function import EvaluateOperatorCostFunction
-                from zquantum.core.circuit import load_circuit_template, load_circuit_template_params, save_circuit_template_params, load_parameter_grid, load_circuit_connectivity
-                from qeopenfermion import load_qubit_operator
-                from zquantum.core.utils import create_object, load_noise_model
-                import json
+                from zquantum.optimizers.utils import optimize_variational_circuit
 
-                ansatz = load_circuit_template('ansatz.json')
-
-                if os.path.isfile('initial_parameters.json'):
-                  initial_parameters = load_circuit_template_params('initial_parameters.json')
-                else:
-                  initial_parameters = None
-
-                # Load qubit op
-                operator = load_qubit_operator('qubitop.json')
-
-                # Load parameter grid
-                if os.path.isfile('parameter_grid.json'):
-                  grid = load_parameter_grid('parameter_grid.json')
-                else:
-                  grid = None
                 optimizer_specs = {{inputs.parameters.optimizer-specs}}
-                if grid is not None and optimizer_specs['function_name'] == 'GridSearchOptimizer':
-                  optimizer = create_object(optimizer_specs, grid=grid)
-                else:
-                  optimizer = create_object(optimizer_specs)
-                  
                 backend_specs = {{inputs.parameters.backend-specs}}
-                if os.path.isfile('noise_model.json'):
-                  backend_specs["noise_model"] = load_noise_model("noise_model.json")
-                if os.path.isfile("device_connectivity.json"):
-                    backend_specs["device_connectivity"] = load_circuit_connectivity(
-                        "device_connectivity.json"
-                    )
-                backend = create_object(backend_specs)
-
-                cost_function = EvaluateOperatorCostFunction(operator, ansatz, backend)
-                if os.path.isfile('constraint_operator.json'):
-                  constraint_operator = load_qubit_operator('constraint_operator.json')
-                  constraint_cost_function = EvaluateOperatorCostFunction(constraint_operator, ansatz, backend)
-                  constraint_functions = ({'type': 'eq', 'fun': constraint_cost_function.evaluate},)
-                  optimizer.constraints = constraint_functions
-
-                opt_results = optimizer.minimize(cost_function, initial_parameters)
-
-                save_optimization_results(opt_results, 'optimization-results.json')
-                save_circuit_template_params(opt_results.opt_params, 'optimized_parameters.json')
+                cost_function_specs = {{inputs.parameters.cost-function-specs}}
```

Comment:
We've discussed it with @simonwa7 and decided this is not the best idea.
Please change it back to the way it was, we have some ideas how to deal with the amount of code in the template, but let's stick to the convention we had so far for now.